operations/jsonnet/microservices/generator.libsonnet

```diff
@@ -1,15 +1,27 @@
 {
   local k = import 'ksonnet-util/kausal.libsonnet',
+
   local container = k.core.v1.container,
   local containerPort = k.core.v1.containerPort,
-  local volumeMount = k.core.v1.volumeMount,
   local deployment = k.apps.v1.deployment,
+  local pvc = k.core.v1.persistentVolumeClaim,
   local volume = k.core.v1.volume,
+  local volumeMount = k.core.v1.volumeMount,
 
   local target_name = 'metrics-generator',
   local tempo_config_volume = 'tempo-conf',
+  local tempo_data_volume = 'metrics-generator-data',
   local tempo_overrides_config_volume = 'overrides',
 
+  tempo_metrics_generator_pvc::
+    pvc.new()
+    + pvc.mixin.metadata.withName(tempo_data_volume)
+    + pvc.mixin.spec.resources.withRequests({ storage: $._config.generator.pvc_size })
+    + pvc.mixin.spec.withAccessModes(['ReadWriteOnce'])
+    + pvc.mixin.spec.withStorageClassName($._config.generator.pvc_storage_class)
+    + pvc.mixin.metadata.withLabels({ app: target_name })
+    + pvc.mixin.metadata.withNamespace($._config.namespace),
```

Comment:
Is there a config to specify that it can be deleted after the pod is deleted? Because
- we won't be replaying the metrics WAL, we only need to prevent filling host disk.
- if not, then the pvc will hang around forever even after the pods are rotated

Reply:
I think there is, but I'll do some research.  Good point.

Reply:
It looks like the reclaim policy is set on the StorageClass, and we aren't able to override it through the PVC, so I think this is already happening for us.

Reply:
I'm not sure I understand. `StorageClass` is the type of disk attached - SSD/etc - and that has a reclaim policy? Then how does the ingester PVC stay around when the pods are rotated? I'm not sure, but if you have checked and confirmed this, I'm good to move forward

Reply:
To answer your question though, the PVC sticks around so that the PV is not deleted.  Depending on the storage class reclaim policy is what happens when the PVC no longer has a claim on the PV.  The `sts` or qdeploy` can be removed independent of the PVC.